cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -471,7 +471,9 @@ func (c *machineController) machineSetNodeGroups() ([]*nodegroup, error) {
 		if err != nil {
 			return err
 		}
-		if ng.MaxSize()-ng.MinSize() > 0 && pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
+		if ng.scalableResource.CanScaleFromZero() {
+			nodegroups = append(nodegroups, ng)
+		} else if ng.MaxSize()-ng.MinSize() > 0 && pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
```

Comment:
we still want to check `ng.MaxSize()-ng.MinSize() > 0` when CanScaleFromZero,  we don't want to add groups where min=max. This will effectively add every machineSet
```
I0323 12:33:09.401145   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1e (min: 0, max: 0, replicas: 0)
I0323 12:33:09.401154   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1f (min: 0, max: 0, replicas: 0)
I0323 12:33:09.401161   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1a (min: 0, max: 0, replicas: 1)
I0323 12:33:09.401168   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1b (min: 0, max: 0, replicas: 1)
I0323 12:33:09.401175   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1c (min: 0, max: 0, replicas: 1)
I0323 12:33:09.401182   21597 clusterapi_provider.go:71] discovered node group: openshift-machine-api/agl230320-cdzht-worker-us-east-1d (min: 0, max: 0, replicas: 0)
```


```
ng.MaxSize()-ng.MinSize() > 0`

Reply:
i.e we should:
```
		if ng.MaxSize()-ng.MinSize() > 0 {
			if ng.scalableResource.CanScaleFromZero() {
				nodegroups = append(nodegroups, ng)
			} else if pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
				nodegroups = append(nodegroups, ng)
			}
		}
```